python/lsst/pipe/tasks/fit_multiband.py

```diff
@@ -0,0 +1,314 @@
+        storageClass="SourceCatalog"
+    )
+
+    def adjustQuantum(self, datasetRefMap):
+        """ Validates the `lsst.daf.butler.DatasetRef` bands against the
+        subtask's list of bands to fit and drops unnecessary bands.
+
+        Parameters
+        ----------
+        datasetRefMap : `NamedKeyDict`
+            Mapping from dataset type to a `set` of
+            `lsst.daf.butler.DatasetRef` objects
+
+        Returns
+        -------
+        datasetRefMap : `NamedKeyDict`
+            Modified mapping of input with possibly adjusted
+            `lsst.daf.butler.DatasetRef` objects.
+
+        Raises
+        ------
+        ValueError
+            Raised if any of the per-band datasets have an inconsistent band
+            set, or if the band set to fit is not a subset of the data bands.
+
+        """
+        datasetRefMap = super().adjustQuantum(datasetRefMap)
+        # Check which bands are going to be fit
+        bands_fit, bands_read_only = self.config.get_band_sets()
+        bands_needed = bands_fit.union(bands_read_only)
+
+        bands_data = None
+        bands_extra = set()
+
+        for type_d, ref_d in datasetRefMap.items():
+            # Datasets without bands in their dimensions should be fine
+            if 'band' in type_d.dimensions:
+                bands_set = {dref.dataId['band'] for dref in ref_d}
+                if bands_data is None:
+                    bands_data = bands_set
+                    if bands_needed != bands_data:
+                        if not bands_needed.issubset(bands_data):
+                            raise ValueError(
+                                f'Datarefs={ref_d} have data with bands in the set={bands_set},'
+                                f'which is not a subset of the required bands={bands_needed} defined by '
+                                f'{self.config.__class__}.fit_multiband='
+                                f'{self.config.fit_multiband._value.__class__}\'s attributes'
+                                f' bands_fit={bands_fit} and bands_read_only()={bands_read_only}.'
+                                f' Add the required bands={bands_needed.difference(bands_data)}.'
+                            )
+                        else:
+                            bands_extra = bands_data.difference(bands_needed)
+                elif bands_set != bands_data:
+                    raise ValueError(
+                        f'Datarefs={ref_d} have data with bands in the set={bands_set}'
+                        f' which differs from the previous={bands_data}); bandsets must be identical.'
+                    )
+                if bands_extra:
+                    for dref in ref_d:
+                        if dref.dataId['band'] in bands_extra:
+                            ref_d.remove(dref)
+        return datasetRefMap
+
+
+class MultibandFitSubConfig(pexConfig.Config):
+    def bands_read_only(self) -> Set:
```

Comment:
I couldn't make this an ABC, but I wanted a default implementation anyway. PyCharm complains that this could be a staticmethod - adding that wouldn't affect subclasses' overrides, would it?